### ISSUE TITLE
Multi OS/Compiler Testing

### ISSUE THREAD

**Issue by astrand130** [astrand130/astrengine#8]
astrengine has been developed on Windows using MSVC. However the engine was developed with the consideration of other Operating Systems and (hopefully) compilers. It has not been tested on other platforms or compilers. There will most likely be issues that need to be fixed and small parts that need to be ported.

A good place to start would be fixing issues on Clang and Creating Precompiled Dependencies for Linux.

**Comment by astrand130** [comment]
PS. Mac Support will not be considered. If someone wants to get it running with MoltenVK go for it. But Apple has made it very clear how they view and treat developers and open standards on their platform. I do not wish to waste my energy with such a platform/company.

**Comment by leetNightshade** [comment]
source/ThirdParty/CMakeLists.txt references xxHash, but the path is xxhash. Fails to compile on BSD. Probably fails on other OS's as well.

**Comment by leetNightshade** [comment]
Why is LibShaderC not included in thirdparty folder, license issue? Too large? 

External dependency of SDL2 is fine, it usually has platform support. But LibShaderC isn't as widespread, little more work to integrate that.

**Comment by astrand130** [comment]
> 
> 
> Why is LibShaderC not included in thirdparty folder, license issue? Too large?
> 
> External dependency of SDL2 is fine, it usually has platform support. But LibShaderC isn't as widespread, little more work to integrate that.

I don't include it now because I work with a pre-combiled version which I distribute separately. Probably a sloppy way of doing it I know. I could put shaderc as a "thirdparty" but the size is a bit big and it is not used directly by the engine runtime, only offline. Will work on it!

> 
> 
> source/ThirdParty/CMakeLists.txt references xxHash, but the path is xxhash (case isn't accurate). Fails to compile on BSD. Probably fails on other OS's as well.

Fixed it! Thank you!

**Comment by astrand130** [comment]
Yeah, looking at the Shaderc code-base again it is a pretty bloated, could easily double the size of the repo (including existing deps). I can make it a git submodule though. For my Physics Library I had to do that as well.

**Comment by astrand130** [comment]
Okay, Never-mind adding Shaderc even as a submodule. It brings Pandora's box of other large Google dependencies. I don't want to drag that into the project.

I will just link to the official pre-compiled versions for Windows/Linux/MacOS in the README instead.

@leetNightshade  I would appreciate it if you could upload the successful build of shaderc for BSD somewhere for future use.
